summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/batch/BatchedStore.scala

```diff
@@ -33,6 +33,23 @@ import cascading.flow.FlowDef
 import org.slf4j.LoggerFactory
 
 import StateWithError.{ getState, putState, fromEither }
+import com.twitter.scalding.serialization.macros.impl.BinaryOrdering
+
+/**
+  * This is the same as scala's Tuple2, except the hashCode is a lazy val.
+  * We do this so when the tuple2 is placed in the map we won't caculate the hash code
+  * unnecessarily several times as the map grows or is transformed.
+  */
+case class LTuple2[T, U](_1: T, _2: U) {
+
+  override lazy val hashCode: Int = scala.runtime.ScalaRunTime._hashCode(this)
```

Comment:
looks like we are always going to hit the hashCode here. Maybe drop lazy and just directly val it?